langwatch/src/optimization_studio/components/properties/BasePropertiesPanel.tsx

```diff
@@ -279,15 +288,15 @@ export function FieldsForm({
   node: Node<Component>;
   field: "parameters" | "inputs" | "outputs";
 }) {
-  const { default_llm, setNode, parameters, setNodeParameter } =
-    useWorkflowStore(
-      useShallow((state) => ({
-        parameters: state.nodes.find((n) => n.id === node.id)?.data.parameters,
-        setNode: state.setNode,
-        default_llm: state.default_llm,
-        setNodeParameter: state.setNodeParameter,
-      }))
-    );
+  const parameters = node.data.parameters;
```

Comment:
I think this is okay to pull it from the destructured object and use it directly. @rogeriochaves ? I assume whatever is passing it in is using the updated node

Reply:
I *think* so, my guess would be that the parent also use the store to retrieve it anyway since this is a panel, I think the issue is only when rendering nodes on the canvas itself, because the node state coming from react-flow might be outdated, but here should be fine